src/ControllerV2/FolderController.php

```diff
@@ -43,6 +46,45 @@ public function list(
         ]);
     }
 
+    #[Route(
+        path: '/beneficiary/{id}/folder/{folderId}/search',
+        name: 'folder_search',
+        requirements: ['id' => '\d+', 'folderId' => '\d+'],
+        methods: ['POST'],
+        condition: 'request.isXmlHttpRequest()',
+    )]
+    #[ParamConverter('folder', class: 'App\Entity\Dossier', options: ['id' => 'folderId'])]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function search(
+        Request $request,
+        Beneficiaire $beneficiary,
+        Dossier $folder,
+        DocumentManager $documentManager,
+        PaginatorService $paginator,
+    ): JsonResponse {
+        $this->denyAccessUnlessGranted('UPDATE', $folder);
+
+        $searchForm = $this->createForm(SearchType::class, null, [
+            'attr' => ['data-controller' => 'ajax-list-filter'],
+            'action' => $this->generateUrl('folder_search', ['id' => $beneficiary->getId(), 'folderId' => $folder->getId()]),
+        ])->handleRequest($request);
+
+        $search = $searchForm->get('search')->getData();
+
+        return new JsonResponse([
+            'html' => $this->renderForm('v2/vault/document/_list.html.twig', [
+                'foldersAndDocuments' => $paginator->create(
+                    $this->isLoggedInUser($beneficiary->getUser())
+                        ? $documentManager->searchFoldersAndDocumentsWithUrl($beneficiary, $search, $folder)
+                        : $documentManager->searchSharedFoldersAndDocumentsWithUrl($beneficiary, $search, $folder),
```

Comment:
Should'nt the document manager handle responsability of wich folders and document to return depending on the logged in user being the beneficiary or not ?